components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRManagementService.java

```diff
@@ -103,8 +109,16 @@ private RegistrationResponseProfile createOAuthApplication(RegistrationRequestPr
 
         //Subscriber's name should be passed as a parameter, since it's under the subscriber
         //the OAuth App is created.
+        String owner =  profile.getOwner();
+        // Replace all unsupported characters
+        String ownerName = owner.replaceAll(DCRConstants.UNSUPPORTED_CHARACTERS_IN_REGISTRY, "_");
+        String applicationName = ownerName + "_" + profile.getClientName();
+        // Regex validation of the application name.
+        if (!DCRMUtils.isRegexValidated(applicationName)) {
+            throw new DCRException("The Application name " + applicationName + " is not valid! It is not adhering to" +
```

Comment:
Use : and space when providing messages for clarity: ex: "The Application name:  " + applicationName